jenkins/PerfCI_Chaos/03_PerfCI_Choas_Windows_VMs_Deployment/Jenkinsfile

```diff
@@ -109,6 +109,33 @@ END
             }
         }
 
+        stage('Prepare Google credential files') {
+            steps {
+                    script {
+                        // Assign the file path to the environment variable
+                        env.GOOGLE_DRIVE_CREDENTIALS = "${env.WORKSPACE}/credentials.json"
+                        env.GOOGLE_DRIVE_TOKEN = "${env.WORKSPACE}/token.json"
+
+                        sh """
+                        echo "Checking if ${env.GOOGLE_DRIVE_CREDENTIALS} exists..."
+                        if [ -f "${env.GOOGLE_DRIVE_CREDENTIALS}" ]; then
+                            echo "File exists. Deleting it..."
+                            rm -f "${env.GOOGLE_DRIVE_CREDENTIALS}"
+                        fi
+                        echo "Checking if ${env.GOOGLE_DRIVE_TOKEN} exists..."
+                        if [ -f "${env.GOOGLE_DRIVE_TOKEN}" ]; then
+                            echo "File exists. Deleting it..."
+                            rm -f "${env.GOOGLE_DRIVE_TOKEN}"
+                        fi
+
+                        echo "Copying secret file to the temporary local file..."
+                        cp "$GOOGLE_DRIVE_CREDENTIALS_FILE" "${env.GOOGLE_DRIVE_CREDENTIALS}"
+                        cp "$GOOGLE_DRIVE_TOKEN_FILE" "${env.GOOGLE_DRIVE_TOKEN}"
```

Comment:
These should not be world readable on general principles; either make sure that the directory containing them is mode 700 or use `umask 77` to make the files themselves unreadable to anyone other than owner.

Reply:
It's good practice to always handle this correctly.  Other users of this might not be on a VPN and might need this protection.

Reply:
I still think you should `umask 77` here.  It's a trivial change but worth doing correctly.